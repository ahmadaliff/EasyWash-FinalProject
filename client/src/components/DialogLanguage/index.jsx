import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { Close } from '@mui/icons-material';
import { Avatar, Dialog, DialogContent, DialogTitle, IconButton, ToggleButton, ToggleButtonGroup } from '@mui/material';
import styled from 'styled-components';

import classes from '@components/DialogLanguage/style.module.scss';

const StyledToggleLanguage = styled(ToggleButton)({
  '&.Mui-selected, &.Mui-selected:hover': {
    color: 'white',
    backgroundColor: '#7ac94c',
    border: 'none',
  },
});

const DialogLanguage = ({ handleSelectLang, locale, open, handleClose }) => (
  <Dialog open={open} onClose={handleClose} fullWidth>
    <DialogTitle fontSize="large">
      <b>
        <FormattedMessage id="app_select_language" />
      </b>
    </DialogTitle>
    <IconButton
      aria-label="close"
      onClick={handleClose}
      sx={{
        position: 'absolute',
        right: 8,
        top: 8,
      }}
    >
      <Close />
    </IconButton>
    <DialogContent>
      <ToggleButtonGroup
        value={locale}
        exclusive
        onChange={handleSelectLang}
        fullWidth
        className={classes.toggleRole}
        size="small"
      >
        <StyledToggleLanguage value="id">
          <Avatar src="/id.png" variant="square" />
          <FormattedMessage id="app_lang_id" />
        </StyledToggleLanguage>
        <StyledToggleLanguage value="en">
          <Avatar src="/en.png" variant="square" />
          <FormattedMessage id="app_lang_en" />
        </StyledToggleLanguage>
      </ToggleButtonGroup>
    </DialogContent>
  </Dialog>
);

DialogLanguage.propTypes = {
  handleSelectLang: PropTypes.func,
  open: PropTypes.bool,
  handleClose: PropTypes.func,
  locale: PropTypes.string.isRequired,
};
export default DialogLanguage;
