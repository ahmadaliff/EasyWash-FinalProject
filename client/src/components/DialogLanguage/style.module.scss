@import '@styles/base';

.toggleRole {
  button {
    @include directionX;
    gap: 0.5rem;
    color: var(--color-primary);
    text-transform: capitalize;
    border-color: var(--color-primary);
  }
}
