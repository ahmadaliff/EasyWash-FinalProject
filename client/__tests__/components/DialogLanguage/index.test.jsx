import { render, fireEvent } from '@utils/testHelper';
import DialogLanguage from '@components/DialogLanguage';

let wrapper;
const mockHandleSelectLang = jest.fn();
const mockHandleClose = jest.fn();

const mockProps = {
  handleSelectLang: mockHandleSelectLang,
  locale: 'en',
  open: true,
  handleClose: mockHandleClose,
};

beforeEach(() => {
  wrapper = render(<DialogLanguage {...mockProps} />);
  jest.clearAllMocks();
});

describe('DialogLanguage Component', () => {
  test('Correct render', () => {
    const { getByText, getByLabelText } = wrapper;
    expect(getByText('Pilih Bahasa')).toBeInTheDocument();
    expect(getByLabelText('close')).toBeInTheDocument();
    expect(getByText('Bahasa Inggris')).toBeInTheDocument();
    expect(getByText('Bahasa Indonesia')).toBeInTheDocument();
  });

  test('Closes the dialog when close button is clicked', () => {
    const { getByLabelText } = wrapper;
    const closeButton = getByLabelText('close');
    fireEvent.click(closeButton);
    expect(mockHandleClose).toHaveBeenCalledTimes(1);
  });

  test('Calls handleSelectLang when a language is selected', () => {
    const { getByText } = wrapper;
    const englishButton = getByText('Bahasa Inggris');
    fireEvent.click(englishButton);
    expect(mockHandleSelectLang).toHaveBeenCalledTimes(1);
  });

  test('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
