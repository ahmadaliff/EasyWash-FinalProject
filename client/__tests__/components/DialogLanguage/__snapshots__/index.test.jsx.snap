// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogLanguage Component Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r3:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-tlc64q-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-1xi8exm-MuiTypography-root-MuiDialogTitle-root"
            id=":r3:"
          >
            <b>
              Pilih Bahasa
            </b>
          </h2>
          <button
            aria-label="close"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-qsfli3-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <div
              class="MuiToggleButtonGroup-root MuiToggleButtonGroup-fullWidth toggleRole css-mgs850-MuiToggleButtonGroup-root"
              role="group"
            >
              <button
                aria-pressed="false"
                class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard sc-bdfCDU gZUEHD MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1xgjc5y-MuiButtonBase-root-MuiToggleButton-root"
                tabindex="0"
                type="button"
                value="id"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-square css-j9wkuc-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="/id.png"
                  />
                </div>
                Bahasa Indonesia
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-pressed="true"
                class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard sc-bdfCDU gZUEHD MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1xgjc5y-MuiButtonBase-root-MuiToggleButton-root"
                tabindex="0"
                type="button"
                value="en"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-square css-j9wkuc-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="/en.png"
                  />
                </div>
                Bahasa Inggris
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
